cfy_manager/components/restservice/restservice.py

```diff
@@ -243,22 +242,20 @@ def _configure_db(self):
             'security_config': REST_SECURITY_CONFIG_PATH
         }
         result = db.check_manager_in_table()
-        if not config[CLUSTER]['active_manager_ip']:
-            if result == constants.DB_NOT_INITIALIZED or config[CLEAN_DB]:
-                logger.info('DB not initialized, creating DB...')
-                db.prepare_db()
-                db.populate_db(configs)
-                config[CLUSTER]['enabled'] = True
-            elif not config[CLEAN_DB]:
-                # Reinstalling the manager with the old DB
-                db.create_amqp_resources(configs)
-        elif result == constants.MANAGER_NOT_IN_DB:
+        if result == constants.MANAGER_NOT_IN_DB:
```

Comment:
the problem is, CLEAN_DB is set also on a first installation, not just if the user wants to recreate the db ( https://github.com/cloudify-cosmo/cloudify-manager-install/blob/master/cfy_manager/main.py#L388 ). Most likely we should change that...

so when the 2nd manager starts, if the `or clean_db` check was first (not elif), it would anyway try to recreate the db. Before, it used to be under a `if active_manager_ip` so the clean_db case was not considered for joining managers at all (as is the case in this new impl)

but i think we should disallow clean_db even if the user DOES want to recreate the db in a cluster, because what will then happen of the other managers in the cluster, they will just be removed from the db and forgotten...

for now i would keep it as is, unless you have some other case in mind, where the manager is not in db but it's not joining a cluster, but i dont think that can happen?

Reply:
I mean, I agree, it's just backwards compatibility we needed to support. 
Personally I would prefer to remove the clean-db option all together and externalize the whole db population phase to a separate script

Reply:
We're on a major version release. Now is the time to remove clean-db...